addons/event_sale/models/event_registration.py

```diff
@@ -26,19 +26,38 @@ class EventRegistration(models.Model):
 
     @api.depends('sale_order_id.state', 'sale_order_id.currency_id', 'sale_order_line_id.price_total')
     def _compute_registration_status(self):
+        confirming_registrations = self.env["event.registration"]
         for so_line, registrations in self.grouped('sale_order_line_id').items():
             cancelled_so_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
             cancelled_so_registrations.state = 'cancel'
             cancelled_registrations = cancelled_so_registrations | registrations.filtered(lambda reg: reg.state == 'cancel')
             if not so_line or float_is_zero(so_line.price_total, precision_digits=so_line.currency_id.rounding):
                 registrations.sale_status = 'free'
-                registrations.filtered(lambda reg: not reg.state or reg.state == 'draft').state = "open"
+                to_open_registrations = registrations.filtered(
```

Comment:
instead of doing 
```python
to_open_registrations.state = "open"
```

can we do

```py
if confirming_registrations:
    confirming_registrations.write({"state": "open"})
```

that way we don't duplicate the code from event

what do you think ?